### PR TITLE
workload: fix --splits regression introduced in #35349

### LINKED ISSUE #37377: roachtest: ycsb/A/nodes=3 failed
SHA: https://github.com/cockroachdb/cockroach/commits/d554884a4e474cc06213230d5ba7d757a88e9e46

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=ycsb/A/nodes=3 PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1279552&tab=buildLog

```
The test failed on branch=release-2.1, cloud=aws:
	cluster.go:1474,ycsb.go:41,cluster.go:1812,errgroup.go:57: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1279552-ycsb-a-nodes-3:4 -- ./workload run ycsb --init --initial-rows=1000000 --splits=100 --workload=A --concurrency=64 --histograms=logs/stats.json --ramp=1m --duration=10m {pgurl:1-3} returned:
		stderr:
		
		stdout:
		, 0x1)
			/usr/local/go/src/runtime/proc.go:302 +0xeb fp=0xc00116bdc0 sp=0xc00116bda0 pc=0x438f8b
		runtime.selectgo(0xc00116bf68, 0xc00116bf48, 0x2, 0xc0014f5840, 0xc0014f5be0)
			/usr/local/go/src/runtime/select.go:313 +0xcc6 fp=0xc00116bf20 sp=0xc00116bdc0 pc=0x448e66
		database/sql.(*DB).connectionResetter(0xc000a24180, 0x19949e0, 0xc000889240)
			/usr/local/go/src/database/sql/sql.go:1014 +0xfb fp=0xc00116bfc8 sp=0xc00116bf20 pc=0x4f89db
		runtime.goexit()
			/usr/local/go/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc00116bfd0 sp=0xc00116bfc8 pc=0x466f61
		created by database/sql.OpenDB
			/usr/local/go/src/database/sql/sql.go:672 +0x193
		bash: line 1:  4171 Aborted                 (core dumped) bash -c "./workload run ycsb --init --initial-rows=1000000 --splits=100 --workload=A --concurrency=64 --histograms=logs/stats.json --ramp=1m --duration=10m 'postgres://root@10.12.33.174:26257?sslmode=disable' 'postgres://root@10.12.36.39:26257?sslmode=disable' 'postgres://root@10.12.34.89:26257?sslmode=disable'"
		Error:  exit status 134
		: exit status 1
	cluster.go:1833,ycsb.go:44,ycsb.go:65,test.go:1251: Goexit() was called

```

### LINKED ISSUE #37378: roachtest: ycsb/A/nodes=3/cpu=32 failed
SHA: https://github.com/cockroachdb/cockroach/commits/d554884a4e474cc06213230d5ba7d757a88e9e46

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=ycsb/A/nodes=3/cpu=32 PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1279552&tab=buildLog

```
The test failed on branch=release-2.1, cloud=aws:
	cluster.go:1474,ycsb.go:41,cluster.go:1812,errgroup.go:57: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1279552-ycsb-a-nodes-3-cpu-32:4 -- ./workload run ycsb --init --initial-rows=1000000 --splits=100 --workload=A --concurrency=64 --histograms=logs/stats.json --ramp=1m --duration=10m {pgurl:1-3} returned:
		stderr:
		
		stdout:
		15ed1e8, 0x0, 0x1809, 0x1)
			/usr/local/go/src/runtime/proc.go:302 +0xeb fp=0xc0007875d8 sp=0xc0007875b8 pc=0x438f8b
		runtime.selectgo(0xc000787768, 0xc000787760, 0x2, 0x0, 0x0)
			/usr/local/go/src/runtime/select.go:313 +0xcc6 fp=0xc000787738 sp=0xc0007875d8 pc=0x448e66
		database/sql.(*DB).connectionOpener(0xc001ce8180, 0x19949e0, 0xc000e23000)
			/usr/local/go/src/database/sql/sql.go:1001 +0xe8 fp=0xc0007877c8 sp=0xc000787738 pc=0x4f88a8
		runtime.goexit()
			/usr/local/go/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc0007877d0 sp=0xc0007877c8 pc=0x466f61
		created by database/sql.OpenDB
			/usr/local/go/src/database/sql/sql.go:671 +0x15d
		bash: line 1:  4488 Aborted                 (core dumped) bash -c "./workload run ycsb --init --initial-rows=1000000 --splits=100 --workload=A --concurrency=64 --histograms=logs/stats.json --ramp=1m --duration=10m 'postgres://root@10.12.35.65:26257?sslmode=disable' 'postgres://root@10.12.47.105:26257?sslmode=disable' 'postgres://root@10.12.45.92:26257?sslmode=disable'"
		Error:  exit status 134
		: exit status 1
	cluster.go:1833,ycsb.go:44,ycsb.go:65,test.go:1251: Goexit() was called

```

### LINKED ISSUE #37379: roachtest: ycsb/B/nodes=3 failed
SHA: https://github.com/cockroachdb/cockroach/commits/d554884a4e474cc06213230d5ba7d757a88e9e46

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=ycsb/B/nodes=3 PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1279552&tab=buildLog

```
The test failed on branch=release-2.1, cloud=aws:
	cluster.go:1474,ycsb.go:41,cluster.go:1812,errgroup.go:57: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1279552-ycsb-b-nodes-3:4 -- ./workload run ycsb --init --initial-rows=1000000 --splits=100 --workload=B --concurrency=64 --histograms=logs/stats.json --ramp=1m --duration=10m {pgurl:1-3} returned:
		stderr:
		
		stdout:
		09, 0x1)
			/usr/local/go/src/runtime/proc.go:302 +0xeb fp=0xc000dbddd8 sp=0xc000dbddb8 pc=0x438f8b
		runtime.selectgo(0xc000dbdf68, 0xc000dbdf60, 0x2, 0xc000474038, 0xc000dbdf78)
			/usr/local/go/src/runtime/select.go:313 +0xcc6 fp=0xc000dbdf38 sp=0xc000dbddd8 pc=0x448e66
		database/sql.(*DB).connectionOpener(0xc00084a0c0, 0x19949e0, 0xc00287d940)
			/usr/local/go/src/database/sql/sql.go:1001 +0xe8 fp=0xc000dbdfc8 sp=0xc000dbdf38 pc=0x4f88a8
		runtime.goexit()
			/usr/local/go/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc000dbdfd0 sp=0xc000dbdfc8 pc=0x466f61
		created by database/sql.OpenDB
			/usr/local/go/src/database/sql/sql.go:671 +0x15d
		bash: line 1:  4172 Aborted                 (core dumped) bash -c "./workload run ycsb --init --initial-rows=1000000 --splits=100 --workload=B --concurrency=64 --histograms=logs/stats.json --ramp=1m --duration=10m 'postgres://root@10.12.35.142:26257?sslmode=disable' 'postgres://root@10.12.37.9:26257?sslmode=disable' 'postgres://root@10.12.42.192:26257?sslmode=disable'"
		Error:  exit status 134
		: exit status 1
	cluster.go:1833,ycsb.go:44,ycsb.go:65,test.go:1251: Goexit() was called

```

### LINKED ISSUE #37380: roachtest: ycsb/B/nodes=3/cpu=32 failed
SHA: https://github.com/cockroachdb/cockroach/commits/d554884a4e474cc06213230d5ba7d757a88e9e46

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=ycsb/B/nodes=3/cpu=32 PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1279552&tab=buildLog

```
The test failed on branch=release-2.1, cloud=aws:
	cluster.go:1474,ycsb.go:41,cluster.go:1812,errgroup.go:57: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1279552-ycsb-b-nodes-3-cpu-32:4 -- ./workload run ycsb --init --initial-rows=1000000 --splits=100 --workload=B --concurrency=64 --histograms=logs/stats.json --ramp=1m --duration=10m {pgurl:1-3} returned:
		stderr:
		
		stdout:
		local/go/src/runtime/proc.go:302 +0xeb fp=0xc0021195c0 sp=0xc0021195a0 pc=0x438f8b
		runtime.selectgo(0xc002119768, 0xc002119748, 0x2, 0x7517000000ffffff, 0x3435393931726573)
			/usr/local/go/src/runtime/select.go:313 +0xcc6 fp=0xc002119720 sp=0xc0021195c0 pc=0x448e66
		database/sql.(*DB).connectionResetter(0xc00057c0c0, 0x19949e0, 0xc001579c40)
			/usr/local/go/src/database/sql/sql.go:1014 +0xfb fp=0xc0021197c8 sp=0xc002119720 pc=0x4f89db
		runtime.goexit()
			/usr/local/go/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc0021197d0 sp=0xc0021197c8 pc=0x466f61
		created by database/sql.OpenDB
			/usr/local/go/src/database/sql/sql.go:672 +0x193
		bash: line 1:  4492 Aborted                 (core dumped) bash -c "./workload run ycsb --init --initial-rows=1000000 --splits=100 --workload=B --concurrency=64 --histograms=logs/stats.json --ramp=1m --duration=10m 'postgres://root@10.12.33.43:26257?sslmode=disable' 'postgres://root@10.12.35.141:26257?sslmode=disable' 'postgres://root@10.12.41.125:26257?sslmode=disable'"
		Error:  exit status 134
		: exit status 1
	cluster.go:1833,ycsb.go:44,ycsb.go:65,test.go:1251: Goexit() was called

```

### LINKED ISSUE #37381: roachtest: ycsb/C/nodes=3/cpu=32 failed
SHA: https://github.com/cockroachdb/cockroach/commits/d554884a4e474cc06213230d5ba7d757a88e9e46

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=ycsb/C/nodes=3/cpu=32 PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1279552&tab=buildLog

```
The test failed on branch=release-2.1, cloud=aws:
	cluster.go:1474,ycsb.go:41,cluster.go:1812,errgroup.go:57: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1279552-ycsb-c-nodes-3-cpu-32:4 -- ./workload run ycsb --init --initial-rows=1000000 --splits=100 --workload=C --concurrency=64 --histograms=logs/stats.json --ramp=1m --duration=10m {pgurl:1-3} returned:
		stderr:
		
		stdout:
		5ed1e8, 0x0, 0x1809, 0x1)
			/usr/local/go/src/runtime/proc.go:302 +0xeb fp=0xc001045dd8 sp=0xc001045db8 pc=0x438f8b
		runtime.selectgo(0xc001045f68, 0xc001045f60, 0x2, 0x0, 0x0)
			/usr/local/go/src/runtime/select.go:313 +0xcc6 fp=0xc001045f38 sp=0xc001045dd8 pc=0x448e66
		database/sql.(*DB).connectionOpener(0xc0006940c0, 0x19949e0, 0xc00112f3c0)
			/usr/local/go/src/database/sql/sql.go:1001 +0xe8 fp=0xc001045fc8 sp=0xc001045f38 pc=0x4f88a8
		runtime.goexit()
			/usr/local/go/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc001045fd0 sp=0xc001045fc8 pc=0x466f61
		created by database/sql.OpenDB
			/usr/local/go/src/database/sql/sql.go:671 +0x15d
		bash: line 1:  4391 Aborted                 (core dumped) bash -c "./workload run ycsb --init --initial-rows=1000000 --splits=100 --workload=C --concurrency=64 --histograms=logs/stats.json --ramp=1m --duration=10m 'postgres://root@10.12.42.216:26257?sslmode=disable' 'postgres://root@10.12.46.42:26257?sslmode=disable' 'postgres://root@10.12.46.214:26257?sslmode=disable'"
		Error:  exit status 134
		: exit status 1
	cluster.go:1833,ycsb.go:44,ycsb.go:65,test.go:1251: Goexit() was called

```

### LINKED ISSUE #37382: roachtest: ycsb/F/nodes=3 failed
SHA: https://github.com/cockroachdb/cockroach/commits/d554884a4e474cc06213230d5ba7d757a88e9e46

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=ycsb/F/nodes=3 PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1279552&tab=buildLog

```
The test failed on branch=release-2.1, cloud=aws:
	cluster.go:1474,ycsb.go:41,cluster.go:1812,errgroup.go:57: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1279552-ycsb-f-nodes-3:4 -- ./workload run ycsb --init --initial-rows=1000000 --splits=100 --workload=F --concurrency=64 --histograms=logs/stats.json --ramp=1m --duration=10m {pgurl:1-3} returned:
		stderr:
		
		stdout:
		 0x1)
			/usr/local/go/src/runtime/proc.go:302 +0xeb fp=0xc000613dc0 sp=0xc000613da0 pc=0x438f8b
		runtime.selectgo(0xc000613f68, 0xc000613f48, 0x2, 0xc000613f68, 0xc000613f60)
			/usr/local/go/src/runtime/select.go:313 +0xcc6 fp=0xc000613f20 sp=0xc000613dc0 pc=0x448e66
		database/sql.(*DB).connectionResetter(0xc0008980c0, 0x19949e0, 0xc000ec15c0)
			/usr/local/go/src/database/sql/sql.go:1014 +0xfb fp=0xc000613fc8 sp=0xc000613f20 pc=0x4f89db
		runtime.goexit()
			/usr/local/go/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc000613fd0 sp=0xc000613fc8 pc=0x466f61
		created by database/sql.OpenDB
			/usr/local/go/src/database/sql/sql.go:672 +0x193
		bash: line 1:  3627 Aborted                 (core dumped) bash -c "./workload run ycsb --init --initial-rows=1000000 --splits=100 --workload=F --concurrency=64 --histograms=logs/stats.json --ramp=1m --duration=10m 'postgres://root@10.12.42.246:26257?sslmode=disable' 'postgres://root@10.12.42.183:26257?sslmode=disable' 'postgres://root@10.12.39.54:26257?sslmode=disable'"
		Error:  exit status 134
		: exit status 1
	cluster.go:1833,ycsb.go:44,ycsb.go:65,test.go:1251: Goexit() was called

```

### LINKED ISSUE #37383: roachtest: ycsb/F/nodes=3/cpu=32 failed
SHA: https://github.com/cockroachdb/cockroach/commits/d554884a4e474cc06213230d5ba7d757a88e9e46

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=ycsb/F/nodes=3/cpu=32 PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1279552&tab=buildLog

```
The test failed on branch=release-2.1, cloud=aws:
	cluster.go:1474,ycsb.go:41,cluster.go:1812,errgroup.go:57: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1279552-ycsb-f-nodes-3-cpu-32:4 -- ./workload run ycsb --init --initial-rows=1000000 --splits=100 --workload=F --concurrency=64 --histograms=logs/stats.json --ramp=1m --duration=10m {pgurl:1-3} returned:
		stderr:
		
		stdout:
		, 0x1)
			/usr/local/go/src/runtime/proc.go:302 +0xeb fp=0xc0008275c0 sp=0xc0008275a0 pc=0x438f8b
		runtime.selectgo(0xc000827768, 0xc000827748, 0x2, 0xc000637380, 0xc000637700)
			/usr/local/go/src/runtime/select.go:313 +0xcc6 fp=0xc000827720 sp=0xc0008275c0 pc=0x448e66
		database/sql.(*DB).connectionResetter(0xc000d7e0c0, 0x19949e0, 0xc00281a040)
			/usr/local/go/src/database/sql/sql.go:1014 +0xfb fp=0xc0008277c8 sp=0xc000827720 pc=0x4f89db
		runtime.goexit()
			/usr/local/go/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc0008277d0 sp=0xc0008277c8 pc=0x466f61
		created by database/sql.OpenDB
			/usr/local/go/src/database/sql/sql.go:672 +0x193
		bash: line 1:  3894 Aborted                 (core dumped) bash -c "./workload run ycsb --init --initial-rows=1000000 --splits=100 --workload=F --concurrency=64 --histograms=logs/stats.json --ramp=1m --duration=10m 'postgres://root@10.12.35.55:26257?sslmode=disable' 'postgres://root@10.12.38.51:26257?sslmode=disable' 'postgres://root@10.12.38.146:26257?sslmode=disable'"
		Error:  exit status 134
		: exit status 1
	cluster.go:1833,ycsb.go:44,ycsb.go:65,test.go:1251: Goexit() was called

```

### LINKED ISSUE #37393: roachtest: tpccbench/nodes=3/cpu=16 failed
SHA: https://github.com/cockroachdb/cockroach/commits/8abb47a1c9795c1463183bc44e776b054bece682

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=tpccbench/nodes=3/cpu=16 PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1279687&tab=buildLog

```
The test failed on branch=master, cloud=aws:
	cluster.go:1833,tpcc.go:853,tpcc.go:554,test.go:1251: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1279687-tpccbench-nodes-3-cpu-16:4 -- ./workload fixtures import tpcc --warehouses=2500 --split --scatter --checks=false {pgurl:1} returned:
		stderr:
		
		stdout:
		0x1 fp=0xc0009577e0 sp=0xc0009577d8 pc=0x466f61
		created by runtime.(*timersBucket).addtimerLocked
			/usr/local/go/src/runtime/time.go:170 +0x114
		
		goroutine 899 [timer goroutine (idle)]:
		runtime.gopark(0x15ed1a8, 0x28cb0c0, 0x3906e301414, 0x1)
			/usr/local/go/src/runtime/proc.go:302 +0xeb fp=0xc000a3e728 sp=0xc000a3e708 pc=0x438f8b
		runtime.goparkunlock(0x28cb0c0, 0xc0009a1414, 0x1)
			/usr/local/go/src/runtime/proc.go:308 +0x53 fp=0xc000a3e758 sp=0xc000a3e728 pc=0x439033
		runtime.timerproc(0x28cb0c0)
			/usr/local/go/src/runtime/time.go:280 +0x288 fp=0xc000a3e7d8 sp=0xc000a3e758 pc=0x456168
		runtime.goexit()
			/usr/local/go/src/runtime/asm_amd64.s:1333 +0x1 fp=0xc000a3e7e0 sp=0xc000a3e7d8 pc=0x466f61
		created by runtime.(*timersBucket).addtimerLocked
			/usr/local/go/src/runtime/time.go:170 +0x114
		bash: line 1:  4273 Aborted                 (core dumped) bash -c "./workload fixtures import tpcc --warehouses=2500 --split --scatter --checks=false 'postgres://root@10.12.41.186:26257?sslmode=disable'"
		Error:  exit status 134
		: exit status 1

```

### PR DESCRIPTION
**Pull request by danhhz**
workload's Table schemas are SQL schemas, but #35349 switched the
initial data to be returned as a coldata.Batch, which has a more limited
set of types. (Or, in the case of simple workloads that return a
[]interface{}, it's roundtripped through coldata.Batch by the `Tuples`
helper.) Notably, this means a SQL STRING column is represented the same
as a BYTES column (ditto UUID, etc).

This caused a regression in splits, which received some []byte data for
a column tried to hand it to SPLIT as a SQL BYTES datum. This didn't
work for the UUID column in tpcc's history table nor the VARCHAR in
ycsb's usertable. Happily, a STRING works for both of these. It also
seems to work for BYTES columns, so it seems like the ambiguity is fine
in this case. When/if someone wants to add a workload that splits a
BYTES primary key column containing non-utf8 data, we'll may need to
revisit.

A more principled fix would be to get the fidelity back by parsing the
SQL schema, which in fact we do in `importccl.makeDatumFromColOffset`.
However, at the moment, this hack works and avoids the complexity and
the undesirable pkg/sql/parser dep.

Closes #37383
Closes #37382
Closes #37381
Closes #37380
Closes #37379
Closes #37378
Closes #37377
Closes #37393

Release note: None

### CODE REVIEW

**Review by tbg — APPROVED** [review]
LGTM but I'd be straight-up lying if I claimed that I really knew what was going on here. I assume you'll be removing the hack not too far down the road, looking forward to more details on this then.

**Comment by danhhz** [comment]
No plans to remove the hack. It would introduce quite a bit of complexity that we don't need until someone tries to use a BYTES column in a primary key of some workload.

**Comment by tbg** [comment]
What will happen when someone does use a bytes column? Will it fail in a hard-to-diagnose way? I'm fine punting this work, but it'd be good to know that the failure mode will be somewhat actionable (will they be told they're in unchartered territory, or at least be able to find your message, basically)

If you have the time, add a bit of color to the commit message before you merge because there's very little context here -- I know something fails around splits, I assume it's because a string portion of a primary key got printed as `fmt.Sprintf(`X'%x'`, x)`, and I assume that that threw off the workload split code which then panicked?

**Comment by danhhz** [comment]
Thanks for pushing me on this, I just took another swing at it that I think is better. RFAL

**Review by tbg — APPROVED** [review]
Thanks, appreciate the second pass!

**Comment by danhhz** [comment]
TFTR!

bors r=tbg

**Pull request merged**
